test/test_data_pet.py

```diff
@@ -52,6 +53,15 @@ def random_dataset(setup_random_pet_data) -> PET:
     )
 
 
+@pytest.fixture
+def random_nifti_file(tmp_path, setup_random_uniform_spatial_data) -> Path:
+    _data, _affine = setup_random_uniform_spatial_data
+    _filename = tmp_path / "random_pet.nii.gz"
+    _img = nb.Nifti1Image(_data, _affine)
+    _img.to_filename(_filename)
+    return _filename
```

Comment:
Haven't had a look at the other modalities tests, but this fixture may be useful in those cases as well.